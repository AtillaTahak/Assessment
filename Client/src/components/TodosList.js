import React from 'react'
import { Todo } from './styled/app.styled'


function TodosList(props) {
    
    return (
        <Todo key={props.data._id} >
            {props.data.title}
        </Todo>
    )
}

export default TodosList
