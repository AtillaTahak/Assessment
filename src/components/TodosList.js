import React from 'react'

function TodosList(props) {
    
    return (
        <div>
            {props.data.title}
        </div>
    )
}

export default TodosList
