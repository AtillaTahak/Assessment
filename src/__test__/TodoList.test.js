import { cleanup, getByDisplayValue, getByRole, getByText, render as rtlRender, screen } from '@testing-library/react';
import TodosList from '../components/TodosList';
import '@testing-library/jest-dom/extend-expect'
import userEvent from '@testing-library/user-event'
import { Provider } from 'react-redux'
import store from '../redux/configureStore'

const render = component => rtlRender(
    <Provider store={store}>
      {component}
    </Provider>
  )

const example = {
    title:"Todo item"
}
test('should list item', () => {
    render(<TodosList data={example}/>)
    const element = screen.getByText(/todo/i);
    expect(element).toBeInTheDocument();
    expect(element).toMatchSnapshot();
    
})
