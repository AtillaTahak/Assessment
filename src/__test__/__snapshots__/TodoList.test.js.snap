// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should list item 1`] = `
<div
  class="sc-gsDKAQ jhGfUW"
>
  Todo item
</div>
`;
